src/ApiService/ApiService/onefuzzlib/ScalesetOperations.cs

```diff
@@ -632,39 +636,37 @@ where x.State.ReadyForReset()
             return;
         }
 
-        var machineIds = new HashSet<Guid>();
+        var nodesToReimage = new List<Node>();
         foreach (var node in nodes) {
             if (node.State != NodeState.Done) {
                 continue;
             }
 
             if (node.DebugKeepNode) {
-                _log.Warning($"not reimaging manually overriden node {node.MachineId:Tag:MachineId} in scaleset {scaleset.ScalesetId:Tag:ScalesetId}");
+                _log.Warning($"not reimaging manually overridden node {node.MachineId:Tag:MachineId} in scaleset {scaleset.ScalesetId:Tag:ScalesetId}");
             } else {
-                _ = machineIds.Add(node.MachineId);
+                nodesToReimage.Add(node);
             }
         }
 
-        if (!machineIds.Any()) {
+        if (!nodesToReimage.Any()) {
             _log.Info($"no nodes to reimage {scaleset.ScalesetId:Tag:ScalesetId}");
             return;
         }
 
         switch (disposalStrategy) {
             case NodeDisposalStrategy.Decommission:
                 _log.Info($"decommissioning nodes");
-                await Async.Task.WhenAll(nodes
-                    .Where(node => machineIds.Contains(node.MachineId))
+                await Async.Task.WhenAll(nodesToReimage
                     .Select(async node => {
                         await _context.NodeOperations.ReleaseScaleInProtection(node).IgnoreResult();
                     }));
                 return;
 
             case NodeDisposalStrategy.ScaleIn:
```

Comment:
I'm a little unclear on the logic of what we are doing here:

- reimage the VMSS node
- release scale-in protection on the VMSS node
- delete our internal knowledge of the node?

Reply:
Yup, that's right. The `Decommission` strategy is what we want to move to (I'm wrapping up validating it in canary). It's going to skip reimaging and just let azure reap our nodes via scale-in. Then instead of deleting the node in our database here, we delete it in `CleanupNodes` (lines 525-530 that I mentioned in the earlier comment).